docs/tutorials/tfx/stub_template.md

```diff
@@ -0,0 +1,108 @@
+
+## Testing the pipeline using Stub Executors
+
+### Introduction
+**You should complete [template.ipynb](https://github.com/tensorflow/tfx/blob/master/docs/tutorials/tfx/template.ipynb) tutorial up to *6* in order to proceed this tutorial.**
+
+This document will provide instructions to test a TensorFlow Extended (TFX) pipeline
+using `BaseStubExecuctor`, which generates fake data using the golden test data. This is intended for users to replace executors they don't want to test so that they could save time from running the actual executors. Stub executor is provided with TFX Python package under `tfx.experimental.pipeline_testing.base_stub_executor`.
+
+This tutorial serves as an extension to `template.ipynb` tutorial, thus you will also use [Taxi Trips dataset](
+https://data.cityofchicago.org/Transportation/Taxi-Trips/wrvz-psew)
+released by the City of Chicago. We strongly encourage you to try modifying the components prior to utilizing stub executors.
+
+### 1. Record the pipeline outputs in Google Cloud Storage
+
+Since this tutorial assumes that you have completed `template.ipynb` up to 6, a successful pipeline run must have been saved in the MLMD, which can be accessed using gRPC server. 
+
+Open a Terminal and run the following commands:
+
+```bash
+gcloud container clusters get-credentials <cluster_name> --zone <compute_zone> --project {GOOGLE_CLOUD_PROJECT}
+```
+
+```bash
+nohup kubectl port-forward deployment/metadata-grpc-deployment -n <namespace> {PORT}:8080 &
+```
+
+Then, you can either:
+
+1.  Open another terminal and clone the tfx GitHub repository.
+ Inside the directory, run the following command:
+```bash
+python tfx/experimental/pipeline_testing/pipeline_recorder.py \
+--output_dir=gs://<project_name>-kubeflowpipelines-default/testdata \
+--host=$host \
+--port=$port \
+--pipeline_name=$pipeline_name
+```
+
+2. Run the python file
+```python
+from tfx.experimental.pipeline_testing import pipeline_recorder_utils
+    pipeline_recorder_utils.record_pipeline(
+        output_dir=gs://<project_name>-kubeflowpipelines-default/testdata,
+        metadata_db_uri=None,
+        host=$host,
+        port=$port,
+        pipeline_name=$pipeline_name,
+        run_id=None)
+```
+
+
+`$output_dir` should be set to a path in Google Cloud Storage where the pipeline outputs are to be recorded, so make sure to replace `<project_name>` with Google Cloud project name.
+
+`$host` and `$port` are hostname and port of the metadata grpc server to connect to MLMD. You may choose any unused port and "localhost" for hostname.
```

Comment:
IIUC, `port` should be the one used for port-forwarding. Could you elaborate more on this?